### PR TITLE
Add OSSEC rule to indicate to admins what ansible-apt-key alert means

### LINKED ISSUE #2156: OSSEC: Running Ansible playbooks triggers level 13 alerts
## Description

Whenever the Ansible playbooks run, alerts like the following fire (here on both app and mon):

```
Received From: (app) 10.20.2.2->/var/log/syslog
Rule: 1003 fired (level 13) -> "Non standard syslog message (size too large)."
Portion of the log(s):

Jul 22 17:06:41 app ansible-apt_key: Invoked with file=None keyserver=None url=None data=-----BEGIN PGP PUBLIC KEY BLOCK-----#012Version: GnuPG v1#012#012mQENBEqg7GsBCACsef8koRT8UyZxiv1Irke5nVpte54TDtTl1za1tOKfthmHbs2I#0124DHWG3qrwGayw+6yb5mMFe0h9Ap9IbilA5a1IdRsdDgViyQQ3kvdfoavFHRxvGON#012tknIyk5Goa36GMBl84gQceRs/4Zx3kxqCV+JYXE9CmdkpkVrh2K3j5+ysDWfD/kO#012dTzwu3WHaAwL8d5MJAGQn2i6bTw4UHytrYemS1DdG/0EThCCyAnPmmb8iBkZlSW8#0126MzVqTrN37yvYWTXk6MwKH50twaX5hzZAlSh9eqRjZLq51DDomO7EumXP90rS5mT#012QrS+wiYfGQttoZfbh3wl5ZjejgEjx+qrnOH7ABEBAAG0JmRlYi50b3Jwcm9qZWN0#012Lm9yZyBhcmNoaXZlIHNpZ25pbmcga2V5iEYEEBECAAYFAkqqojIACgkQ61qJaiiY#012i/WmOgCfTyf3NJ7wHTBckwAeE4MSt5ZtXVsAn0XDq8PWWnk4nK6TlevqK/VoWItF#012iEYEEBECAAYFAkqsYDUACgkQO50JPzGwl0voJwCcCSokiJSNY+yIr3nBPN/LJldb#012xekAmwfU60GeaWFwz7hqwVFL23xeTpyniEYEEBECAAYFAkt9ndgACgkQYhWWT1sX#012KrI5TACfcBPbsaPA1AUVVXXPv0KeWFYgVaIAoMr3jwd1NYVD6Te3D+yJhGzzCD6P#012iEYEEBECAAYFAkt+li8ACgkQTlMAGaGhvAU4FwCfX3H4Ggm/x0yIAvmt4CW8AP9F#0125D8AoKapuwbjsGncT3UdNFiHminAaq1tiEYEEBECAAYFAky6mjsACgkQhfcmMSeh#012yJpL+gCggxs4C5o+Oznk7WmFrPQ3lbnfDKIAni4p20aRuwx6QWGH8holjzTSmm5F#012iEYEEBECAAYFAlMI0FEACgkQhEMxewZV94DLagCcDG5SR00+00VHzBVE6fDg027e#012N2sAnjNLOYbRSBxBnELUDKC7Vjaz/sAMiEwEExECAAwFAkqg7nQFgwll/3cACgkQ#0123nqvbpTAnH+GJA

```

```
Received From: mon->/var/log/syslog
Rule: 1003 fired (level 13) -> "Non standard syslog message (size too large)."
Portion of the log(s):

Jul 22 17:06:42 mon ansible-apt_key: Invoked with file=None keyserver=None url=None data=-----BEGIN PGP PUBLIC KEY BLOCK-----#012Version: GnuPG v1#012#012mQENBEqg7GsBCACsef8koRT8UyZxiv1Irke5nVpte54TDtTl1za1tOKfthmHbs2I#0124DHWG3qrwGayw+6yb5mMFe0h9Ap9IbilA5a1IdRsdDgViyQQ3kvdfoavFHRxvGON#012tknIyk5Goa36GMBl84gQceRs/4Zx3kxqCV+JYXE9CmdkpkVrh2K3j5+ysDWfD/kO#012dTzwu3WHaAwL8d5MJAGQn2i6bTw4UHytrYemS1DdG/0EThCCyAnPmmb8iBkZlSW8#0126MzVqTrN37yvYWTXk6MwKH50twaX5hzZAlSh9eqRjZLq51DDomO7EumXP90rS5mT#012QrS+wiYfGQttoZfbh3wl5ZjejgEjx+qrnOH7ABEBAAG0JmRlYi50b3Jwcm9qZWN0#012Lm9yZyBhcmNoaXZlIHNpZ25pbmcga2V5iEYEEBECAAYFAkqqojIACgkQ61qJaiiY#012i/WmOgCfTyf3NJ7wHTBckwAeE4MSt5ZtXVsAn0XDq8PWWnk4nK6TlevqK/VoWItF#012iEYEEBECAAYFAkqsYDUACgkQO50JPzGwl0voJwCcCSokiJSNY+yIr3nBPN/LJldb#012xekAmwfU60GeaWFwz7hqwVFL23xeTpyniEYEEBECAAYFAkt9ndgACgkQYhWWT1sX#012KrI5TACfcBPbsaPA1AUVVXXPv0KeWFYgVaIAoMr3jwd1NYVD6Te3D+yJhGzzCD6P#012iEYEEBECAAYFAkt+li8ACgkQTlMAGaGhvAU4FwCfX3H4Ggm/x0yIAvmt4CW8AP9F#0125D8AoKapuwbjsGncT3UdNFiHminAaq1tiEYEEBECAAYFAky6mjsACgkQhfcmMSeh#012yJpL+gCggxs4C5o+Oznk7WmFrPQ3lbnfDKIAni4p20aRuwx6QWGH8holjzTSmm5F#012iEYEEBECAAYFAlMI0FEACgkQhEMxewZV94DLagCcDG5SR00+00VHzBVE6fDg027e#012N2sAnjNLOYbRSBxBnELUDKC7Vjaz/sAMiEwEExECAAwFAkqg7nQFgwll/3cACgkQ#0123nqvbpTAnH+GJA
```

## Suggested action

The presence of an alert here is actually useful - they alert an admin when a change in vars occurred or some other change was pushed to the server. In the model where there are several admins, this is extremely useful as it means they know that they need to e.g. SSH into the server or talk to their fellow admin to determine what changes were made (see #2096). 

With that said, for some admins it's unclear what the above means. A rule needs to be written to parse this `syslog` event and produce an understandable `description`, e.g. "Ansible playbook run on server".

## User Stories

As a SecureDrop administrator, I want to know when the Ansible playbooks are run so I am apprised of any changes to the system configuration.

### LINKED ISSUE #2182: Add developer docs for testing OSSEC rules
As of #2137 we have tests for OSSEC rules, but we should write a document explaining how to test OSSEC rules and how to write `testinfra` tests for them.

### PR DESCRIPTION
**Pull request by redshiftzero**
## Status

Ready for review

## Description of Changes

This PR:

* Towards #2156. This adds an OSSEC rule to catch the alerts that bubble up when the Ansible playbooks are run and allow the alert to fire with a useful description for admins.
* DRYs up the OSSEC configuration tests and moves alert text to the YAML vars.
* Closes #2182.

## Testing

- [ ] Config tests fail on 1da1664 in the staging environment (error should involve `rule_id`)
- [ ] After 7d8af79 all config tests pass

## Deployment

Rules should be deployed via the SecureDrop OSSEC deb packages without issue.

### If you made changes to the system configuration:

- [x] [Configuration tests](https://docs.securedrop.org/en/latest/development/testing_configuration_tests.html) pass

### If you made changes to documentation:

- [x] Doc linting passed locally

### CODE REVIEW

**Comment by conorsch** [comment]
Reviewing now.

File: testinfra/vars/mon-staging.yml [hunk 1]
```diff
@@ -13,3 +13,75 @@ tor_services:
...
+      eTpyniEYEEBECAAYFAkt9ndgACgkQYhWWT1sX#012KrI5TACfcBPbsaPA1A
+      UVVXXPv0KeWFYgVaIAoMr3jwd1NYVD6Te3D+yJhGzzCD6P#012iEYEEBECA
+      AYFAkt+li8ACgkQTlMAGaGhvAU4FwCfX3H4Ggm/x0yIAvmt4CW8AP9F#012
+      5D8AoKapuwbjsGncT3UdNFiHminAaq1tiEYEEBECAAYFAky6mjsACgkQhfc
+      mMSeh#012yJpL+gCggxs4C5o+Oznk7WmFrPQ3lbnfDKIAni4p20aRuwx6QW
+      GH8holjzTSmm5F#012iEYEEBECAAYFAlMI0FEACgkQhEMxewZV94DLagCcD
+      G5SR00+00VHzBVE6fDg027e#012N2sAnjNLOYbRSBxBnELUDKC7Vjaz/sAM
+      iEwEExECAAwFAkqg7nQFgwll/3cACgkQ#0123nqvbpTAnH+GJA
+    level: "13"
+    rule_id: "400001"
```
Comment (conorsch) [inline testinfra/vars/mon-staging.yml]: Beautiful job on the new test vars! The additions even pass YAML linting standards. 🥇

File: docs/development/updating_ossec.rst [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+Deployment
+----------
+
+The OSSEC rules and associated configuration files are distributed via Debian
+packages maintained by Freedom of the Press Foundation. Any changes made to OSSEC
+configuration files will land on production SecureDrop monitoring servers as
+part of each SecureDrop release. This upgrade will occur automatically.
+
+.. note:: This also means that any changes made locally by administrators to
+          their OSSEC rules will not persist after a SecureDrop update.
```
Comment (conorsch) [inline docs/development/updating_ossec.rst]: In the `note::`, the leading "This" refers to statements made in the preceding paragraph text, which somewhat defeats the purpose of using a Note. How about: 

> The use of automatic upgrades for release deployment means that any changes made locally by administrators to their OSSEC rules will not persist after a SecureDrop update.

Comment (redshiftzero) [reply docs/development/updating_ossec.rst]: Fair! Let me know when you're done reviewing / when it's cool for me to force push without being annoying and I'll fix this.

**Comment by conorsch** [comment]
Confirmed that the testinfra tests fail when run against staging machines provisioned on latest `develop`, and pass when run against staging machines provisioned with new deb packages based on this feature branch.

The docs additions are solid. We'll certainly have to flesh out the docs more in the coming releases—some of the info such as useful logfiles should be ported from the OSSEC guide to the writing alerts guide—but these additions are great to have in as soon as possible. 

One question: won't only the _first_ Ansible-related alert get a custom message, and all the other events will continue to fire? It appears as if _only_ the apt_key module events will get a custom message, and all the rest will operate as before. If that's the case, then perhaps we should leave #2156 open, since it's still very much a problem. 

For example, during testing just now, running the playbooks against staging creates dozens of alerts—if we detect an Ansible playbook run, I'd prefer to silence the subsequent alerts, as well.

**Comment by redshiftzero** [comment]
What you say re: this PR only addressing the first Ansible-related alert is true. I agree in that case that #2156 should stay open until we silence the other alerts (if they are not providing useful information). Edited PR description accordingly.

**Comment by conorsch** [comment]
Excellent, thanks for the clarification, @redshiftzero.

**Comment by redshiftzero** [comment]
FYI I made that docs change @conorsch

**Pull request merged**